spec/javascripts/unit/modules/landing-page-map.spec.js

```diff
@@ -0,0 +1,352 @@
+describe('Map Block module', function () {
+  'use strict'
+
+  var el, module
+
+  function createMapDom (apiKey) {
+    el = document.createElement('div')
+    el.innerHTML = `
+      <div id="mapfilter-hint" class="govuk-hint">Filter by</div>
+        <div class="govuk-checkboxes">
+          <div class="govuk-checkboxes__item">
+            <input type="checkbox" name="favourite_small_synonym[]" id="mapfilter-0" value="cdc" class="govuk-checkboxes__input" checked=""><label for="mapfilter-0" class="govuk-label govuk-checkboxes__label">Community Diagnostic Centres</label>
+          </div>
+          <div class="govuk-checkboxes__item">
+            <input type="checkbox" name="favourite_small_synonym[]" id="mapfilter-1" value="sh" class="govuk-checkboxes__input" checked=""><label for="mapfilter-1" class="govuk-label govuk-checkboxes__label">Surgical hubs</label>
+          </div>
+        </div>
+      </fieldset>
+      <div class="map__canvas"
+        id="map"
+        data-api-key="${apiKey}"
+        data-icon-cdc-default="landing_page/map/CDC-symbol-default.svg"
+        data-icon-cdc-active="landing_page/map/CDC-symbol-active.svg"
+        data-icon-sh-default="landing_page/map/SH-symbol-default.svg"
+        data-icon-sh-active="landing_page/map/SH-symbol-active.svg"
+        data-module="ga4-event-tracker">
+      </div>
+    `
+    document.body.appendChild(el)
+  }
+
+  function agreeToCookies () {
+    GOVUK.setCookie('cookies_policy', '{"essential":true,"settings":true,"usage":true,"campaigns":true}')
+  }
+
+  function denyCookies () {
+    GOVUK.setCookie('cookies_policy', '{"essential":false,"settings":false,"usage":false,"campaigns":false}')
+  }
+
+  afterEach(function () {
+    document.body.removeChild(el)
+  })
+
+  describe('when initialising the map', function () {
+    function setupMap (apiKey) {
+      denyCookies()
+      createMapDom(apiKey)
+      module = new GOVUK.Modules.LandingPageMap(el.querySelector('.map__canvas'))
+      // need to spy on all these functions so they don't call through and error
+      spyOn(module, 'initialiseMap')
+      spyOn(module, 'enableTracking')
+      spyOn(module, 'addOverlays')
+      module.init()
+    }
+
+    it('does nothing if there is no API key', function () {
+      setupMap('') // need to pass an empty string, otherwise becomes the string 'undefined'
```

Comment:
You could have a default value in the function definition instead of passing an empty string. E.g. `function setupMap (apiKey = '') {`

Reply:
Good shout, fixed. Also realised for the other version of this function I don't need to keep passing a fake apiKey as it never changes in that test.